sxw_soilwat.c

```diff
@@ -243,43 +266,26 @@ static void _update_productivity(void) {
             ForEachGroup(g) {
                 if (1 == RGroup[g]->veg_prod_type) { //tree
                     v->veg[SW_TREES].pct_live[m] += _prod_pctlive[Igp(g, m)] * RGroup[g]->rgroupFractionOfVegTypeBiomass;
-                    v->veg[SW_TREES].biomass[m] += _prod_bmass[Igp(g, m)] * bmassg[g];
+                    v->veg[SW_TREES].biomass[m] += _prod_bmass[Igp(g, m)] * bmassg[g] / v->veg[SW_TREES].cov.fCover;
                 } else if (2 == RGroup[g]->veg_prod_type) { //shrub
                     v->veg[SW_SHRUB].pct_live[m] += _prod_pctlive[Igp(g, m)] * RGroup[g]->rgroupFractionOfVegTypeBiomass;
-                    v->veg[SW_SHRUB].biomass[m] += _prod_bmass[Igp(g, m)] * bmassg[g];
+                    v->veg[SW_SHRUB].biomass[m] += _prod_bmass[Igp(g, m)] * bmassg[g] / v->veg[SW_SHRUB].cov.fCover;
                 } else if (3 == RGroup[g]->veg_prod_type) { //grass
                     v->veg[SW_GRASS].pct_live[m] += _prod_pctlive[Igp(g, m)] * RGroup[g]->rgroupFractionOfVegTypeBiomass;
-                    v->veg[SW_GRASS].biomass[m] += _prod_bmass[Igp(g, m)] * bmassg[g];
+                    v->veg[SW_GRASS].biomass[m] += _prod_bmass[Igp(g, m)] * bmassg[g] / v->veg[SW_GRASS].cov.fCover;
                 } else if (4 == RGroup[g]->veg_prod_type) { //forb
                     v->veg[SW_FORBS].pct_live[m] += _prod_pctlive[Igp(g, m)] * RGroup[g]->rgroupFractionOfVegTypeBiomass;
-                    v->veg[SW_FORBS].biomass[m] += _prod_bmass[Igp(g, m)] * bmassg[g];
+                    v->veg[SW_FORBS].biomass[m] += _prod_bmass[Igp(g, m)] * bmassg[g] / v->veg[SW_FORBS].cov.fCover;
                 }
             }
 
-            v->veg[SW_TREES].litter[m] = (vegTypeBiomass[0] * _prod_litter[m]);
-            v->veg[SW_SHRUB].litter[m] = (vegTypeBiomass[1] * _prod_litter[m]);
-            v->veg[SW_GRASS].litter[m] = (vegTypeBiomass[2] * _prod_litter[m]);
-            v->veg[SW_FORBS].litter[m] = (vegTypeBiomass[3] * _prod_litter[m]);
-        }
-    }
-
-    if (GT(totbmass, 0.)) {
-        //if (ZRO(biomass))
-        //	biomass = 1;
-        v->veg[SW_TREES].cov.fCover = (vegTypeBiomass[0] / totbmass);
-        v->veg[SW_SHRUB].cov.fCover = (vegTypeBiomass[1] / totbmass);
-        v->veg[SW_GRASS].cov.fCover = (vegTypeBiomass[2] / totbmass);
-        v->veg[SW_FORBS].cov.fCover = (vegTypeBiomass[3] / totbmass);
-        //TODO: figure how to calculate bareground fraction.
-        v->bare_cov.fCover = 0;
-    } else {
-
-        ForEachVegType(k) {
-            v->veg[k].cov.fCover = 0.0;
+            v->veg[SW_TREES].litter[m] = (vegTypeBiomass[0] / v->veg[SW_TREES].cov.fCover * _prod_litter[m]);
+            v->veg[SW_SHRUB].litter[m] = (vegTypeBiomass[1] / v->veg[SW_SHRUB].cov.fCover * _prod_litter[m]);
+            v->veg[SW_GRASS].litter[m] = (vegTypeBiomass[2] / v->veg[SW_GRASS].cov.fCover * _prod_litter[m]);
+            v->veg[SW_FORBS].litter[m] = (vegTypeBiomass[3] / v->veg[SW_FORBS].cov.fCover * _prod_litter[m]);
         }
```

Comment:
same point as above - grasses and forbs switched?

Also, why not just use` v->veg[SW_SHRUB].biomass[m]` here instead of `vegTypeBiomass[1] / v->veg[SW_SHRUB].cov.fCover`

Reply:
@dschlaep - do you still think it best to re-calculatd vegtype biomass here rather than using the previous calculation of it above?

Reply:
@kpalmqui Sorry, I overlooked your second comment about how to calculate litter. 

The code was 
```
v->veg[SW_TREES].litter[m] = (vegTypeBiomass[0] * _prod_litter[m]);
```
and I changed it with this branch to
```
v->veg[SW_TREES].litter[m] = (vegTypeBiomass[0] / v->veg[SW_TREES].cov.fCover * _prod_litter[m]);
```
which is equivalent to
```
v->veg[SW_TREES].litter[m] = totbmass * _prod_litter[m];
```
--> This does not make sense!

I agree that your suggestion makes sense
```
v->veg[SW_TREES].litter[m] = v->veg[SW_TREES].biomass[m] * _prod_litter[m];
```
and is more in the spirit of what it used to be. I'll fix it.

It also makes clear that I was wrong in my argument here https://github.com/DrylandEcology/SOILWAT2/pull/264#discussion_r302965106 ... sorry!

Reply:
@kpalmqui However, this also highlights the question why we use the same `_prod_litter` proportions for every vegetation type: woody types seem to behave differently from herbaceous ones? Maybe this could be a future enhancement to vary `_prod_litter` inputs by vegetation type?

Reply:
@kpalmqui I'm changing my mind... 
```
v->veg[SW_TREES].litter[m] = (vegTypeBiomass[0] / v->veg[SW_TREES].cov.fCover * _prod_litter[m]);
```
was correct (well, the simplified from of `v->veg[SW_TREES].litter[m] = totbmass * _prod_litter[m];`).

The fix to `v->veg[SW_TREES].litter[m] = v->veg[SW_TREES].biomass[m] * _prod_litter[m];` is incorrect because `v->veg[SW_TREES].biomass[m] * _prod_litter[m]` represents the amount of `SW_TREE`-litter in units of grams per square-meter given the cover fraction of trees -- instead of what `SOILWAT2` needs, i.e., grams in square-meter for 100% cover, i.e., divide by `v->veg[SW_TREES].cov.fCover` ...

Given that `_prod_litter` is the same for each vegetation type, the calculation `totbmass * _prod_litter[m]` is the correct one, i.e., it is the only one that produces correct units. However, it really highlights https://github.com/DrylandEcology/STEPWAT2/issues/349

@kpalmqui What do you think: should I "fix" it to `totbmass * _prod_litter[m]`?

Reply:
@dschlaep if I am understanding correctly, you are proposed to calculate litter as `totbmass * _prod_litter[m]` for each veg type until a solution to #349 is documented (i.e. unique `prod_litter` values are dervied)? 

Wouldn't that be simulating way too much litter? Perhaps I am not understanding...

Reply:
@kpalmqui  oh, my! I was in a confused state when I wrote that yesterday indeed... `v->veg[SW_TREES].biomass[m]` is scaled correctly -- we just calculated it that way ..., so: `v->veg[k].litter[m] = v->veg[k].biomass[m] * _prod_litter[m];` as it currently is, seems to be correct. Sorry for my confusion!

Reply:
@dschlaep no problem at all! It looks like we are ready to merge this into master?